walk-in-interview/src/main/webapp/maps.js

```diff
@@ -54,4 +54,50 @@ function addMarker(map, job) {
   });
 }
 
-export {createMap, addMarker, JOB_MAP_ZOOM};
+/**
+ * Finds the coordinates based on the Singapore postal code.
+ *
+ * @param {String} postalCode The postal code.
+ * @return {Promise} With the coordinates.
+ */
+function findCoordinates(postalCode) {
+  const request = {
+    query: `Singapore ${postalCode}`,
+    fields: ['name', 'geometry'],
+  };
+
+  /* HTML element required for rendering the results. */
+  const service = new google.maps.places.PlacesService(
+      document.getElementById('places-results'));
+
+  return new Promise((resolve, reject) => {
+    service.findPlaceFromQuery(request, (results, status) => {
+      if (status === google.maps.places.PlacesServiceStatus.OK) {
+        // there should only be one location with that postal code
+        if (results.length != 1) {
+          throw new Error('unable to find one place for given postal code: ' +
+            postalCode);
+        }
+
+        const location = results[0].geometry.location;
+        const latitude = location.lat();
+        const longitude = location.lng();
+
+        // rectangular bounds for Singapore
+        if (latitude > 1.4775 || latitude < 1.1356 ||
+            longitude > 104.1215 || longitude < 103.5582) {
+          throw new Error('coordinates should be in Singapore');
```

Comment:
do u need to handle this error separately? e.g. show different error message?

Reply:
I don't think we need to show the error msg to the user because if this occurred, the error should not be their fault

Reply:
but shall we inform the user? in case they feel confusing what's wrong with the page

Reply:
ok sure, it'll show an error in the postal code field